PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = 'hr.employee'
-    _inherit = 'hr.resume.line'
-    _inherit = 'slide.channel.partner'
-    _inherit = 'slide.channel'

Total Changes: 15

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/hr_skills_slides/models/hr_employee.py
+++ b/addons/hr_skills_slides/models/hr_employee.py
@@ -4,8 +4,8 @@
 from odoo import api, fields, models, _
 
 
-class Employee(models.Model):
-    _inherit = 'hr.employee'
+class HrEmployee(models.Model):
+    _inherit = ['hr.employee']
 
     subscribed_courses = fields.Many2many('slide.channel', related='user_partner_id.slide_channel_ids')
     has_subscribed_courses = fields.Boolean(compute='_compute_courses_completion_text', groups="hr.group_hr_user,base.group_system")

--- a/addons/hr_skills_slides/models/hr_resume_line.py
+++ b/addons/hr_skills_slides/models/hr_resume_line.py
@@ -4,8 +4,8 @@
 from odoo import fields, models, api
 
 
-class ResumeLine(models.Model):
-    _inherit = 'hr.resume.line'
+class HrResumeLine(models.Model):
+    _inherit = ['hr.resume.line']
 
     display_type = fields.Selection(selection_add=[('course', 'Course')])
     channel_id = fields.Many2one('slide.channel', string="Course", readonly=True, index='btree_not_null')

--- a/addons/hr_skills_slides/models/slide_channel.py
+++ b/addons/hr_skills_slides/models/slide_channel.py
@@ -8,7 +8,7 @@ from odoo.tools import html2plaintext
 
 
 class SlideChannelPartner(models.Model):
-    _inherit = 'slide.channel.partner'
+    _inherit = ['slide.channel.partner']
 
     def _recompute_completion(self):
         res = super(SlideChannelPartner, self)._recompute_completion()
@@ -64,8 +64,9 @@ class SlideChannelPartner(models.Model):
                 })
                 self.env.user.employee_id.message_post(body=msg)
 
-class Channel(models.Model):
-    _inherit = 'slide.channel'
+
+class SlideChannel(models.Model):
+    _inherit = ['slide.channel']
 
     def _action_add_members(self, target_partners, member_status='joined', raise_on_access=False):
         res = super()._action_add_members(target_partners, member_status=member_status, raise_on_access=raise_on_access)
